internal/pkg/template/templates/environment/cf.yml

```diff
@@ -104,7 +108,28 @@ Resources:
       Tags:
         - Key: Name
           Value: !Sub 'copilot-${AppName}-${EnvironmentName}-lb'
-  # Only accept requests coming from the public ALB or other containers in the same security group.
+  InternalLoadBalancerSecurityGroup:
+    Metadata:
+      'aws:copilot:description': 'A security group for your internal load balancer allowing HTTP traffic from within the VPC'
+    Condition: CreateInternalALB
+    Type: AWS::EC2::SecurityGroup
+    Properties:
+      GroupDescription: Access to the internal load balancer
+      SecurityGroupIngress:
+        - CidrIp: 0.0.0.0/0
+          Description: Allow from anyone within the VPC on port 80
+          FromPort: 80
+          IpProtocol: tcp
+          ToPort: 80
```

Comment:
I wonder if this is what we want to do (anything in the VPC being able to make requests to the load balancer) vs. only allowing resources with the `EnvironmentSecurityGroup` associated to make requests to the LB.

So just like how you have `EnvironmentSecurityGroupIngressFromInternalALB` there would be another one called `InternalALBIngressFromEnvironmentSecurityGroup`.

Would you mind reaching out to the various internal ALB github issues and asking them if they would want only their copilot services to make requests to the ALB or anything in the VPC to have access to it?

Reply:
I know that a user mentioned that the ALB should be accessible from any instance in the VPC but I think we should start from least-privileged and then expand the scope of this security group. What do you think?

So let's start  with only allowing ingress only from the Environment Security Group. With the environment manifest around the corner, we'll allow folks to customize the security group.
If other clients also want to increase the scope, we can then decide to output the `InternalLoadBalancerSecurityGroup` so that they can write a  `AWS::EC2::SecurityGroupIngress` addon